src/mlpack/core/tree/rectangle_tree/x_tree_split_impl.hpp

```diff
@@ -645,25 +601,30 @@ bool XTreeSplit::SplitNonLeafNode(TreeType *tree,std::vector<bool>& relevels)
     }
   }
 
-  std::vector<sortStruct<ElemType>> sorted(tree->NumChildren());
+  std::vector<std::pair<ElemType, size_t>> sorted(tree->NumChildren());
   if (lowIsBest)
   {
     for (size_t i = 0; i < sorted.size(); i++)
     {
-      sorted[i].d = tree->Child(i).Bound()[bestAxis].Lo();
-      sorted[i].n = i;
+      sorted[i].first = tree->Child(i).Bound()[bestAxis].Lo();
+      sorted[i].second = i;
     }
   }
   else
   {
     for (size_t i = 0; i < sorted.size(); i++)
     {
-      sorted[i].d = tree->Child(i).Bound()[bestAxis].Hi();
-      sorted[i].n = i;
+      sorted[i].first = tree->Child(i).Bound()[bestAxis].Hi();
+      sorted[i].second = i;
     }
   }
 
-  std::sort(sorted.begin(), sorted.end(), structComp<ElemType>);
+  std::sort(sorted.begin(), sorted.end(),
+      [] (const std::pair<ElemType, size_t>& p1,
+          const std::pair<ElemType, size_t>& p2)
+      {
+        return p1.first < p2.first;
+      });
```

Comment:
Do you think it would be better to define this comparator function in some header file in the rectangle_tree directory instead of writing the lambda many times?  If not, I'll merge as-is, but it might be a good idea to make the code a bit shorter.